internal/impl/mongodb/bloblang.go

```diff
@@ -0,0 +1,38 @@
+package mongodb
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/benthosdev/benthos/v4/public/bloblang"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func init() {
+	objectIdParseSpec := bloblang.NewPluginSpec().
+		Beta().
+		Category("Parsing").
+		Description("Decodes a string or a date into a [MongoDB's ObjectId](https://www.mongodb.com/docs/manual/reference/method/ObjectId/).").
+		Example("", `root._id = content().parse_object_id()`)
+
+	if err := bloblang.RegisterMethodV2(
+		"parse_object_id", objectIdParseSpec,
```

Comment:
The naming here should probably be prefixed with something like `mongodb_` in order to make it clear that this is very specific to mongodb use cases